ui/src/__tests__/spec/wdio.conf.js

```diff
@@ -1,73 +1,145 @@
-          );
-      });
+let sdAthenzKeyFilePath = WORK_DIR + '/func.key.pem';
+let sdAthenzCertFiledPath = WORK_DIR + '/func.cert.pem';
+let localAthenzKeyFilePath = '~/.athenz/keys/' + athenzDomain + '.' + athenzService + '.key.pem';
+let localAthenzCertFilePath = '~/.athenz/certs/' + athenzDomain + '.' + athenzService + '.cert.pem';
 
-const config = {
+let functionalConfig = {
+  'athensZtsAPI': appConfig.ztsOnPrem,
+  'athenzService': athenzService,
+  'athenzDomain': athenzDomain,
+  'athenzKeyFile': sdv4FunctionalJob ? sdAthenzKeyFilePath : localAthenzKeyFilePath,
+  'athenzCertFile': sdv4FunctionalJob ? sdAthenzCertFiledPath : localAthenzCertFilePath,
+  'sauceUser': sdv4FunctionalJob ? appConfig.sauceUser : process.env.SAUCE_USERNAME,
+  'sauceKey': process.env.SAUCE_KEY,
+  'instance': process.env.INSTANCE || 'https://local-ui.athenz.ouryahoo.com/',
+  'cookieDomain': process.env.COOKIE || appConfig.cookieDomain,
+  'sauceSeleniumAddress': appConfig.sauceSeleniumAddress,
+  'screenResolution': '1600x1200'
+};
+
+const sauceLabsKey = functionalConfig.sauceKey || '';
+const sauceLabsUser = functionalConfig.sauceUser || '';
+const localOrRemote = { };
+if (!sauceLabsUser) {
+  //
+  // Test runner services
+  // Services take over a specific job you don't want to take care of. They enhance
+  // your test setup with almost no effort. Unlike plugins, they don't add new
+  // commands. Instead, they hook themselves up into the test process.
+  localOrRemote.saucelabs = {
+    runner: 'local',
+    services: ['chromedriver']
+  };
+  localOrRemote.capabilities = [
+    {
+      browserName: 'chrome',
+      browserVersion: 'latest',
+      acceptInsecureCerts: true
+    }
+  ];
+} else {
+  // SauceLabs Settings
+  localOrRemote.saucelabs = {
+    user: sauceLabsUser,
+    key: sauceLabsKey,
+    services: ['sauce'],
+    sauceConnect: true,
+    region: 'us',
+    updateJob: true,
+    sauceSeleniumAddress: functionalConfig.sauceSeleniumAddress,
+    setJobName: (config, capabilities, suiteTitle) => {
+      capabilities.name = suiteTitle + ':' + 'Athenz UI Tests';
+    }
+  };
+  // Capabilities - These are the browsers to use in SauceLabs
+  localOrRemote.capabilities = [
+    {
+      browserName: 'chrome',
+      acceptInsecureCerts: true,
+      browserVersion: 'latest',
+      platformName: 'OS X 12',
+      maxInstances: 7,
+      'sauce:options': {
+        tunnelIdentifier: TUNNEL_IDENTIFIER + '-' + (Math.floor(Math.random() * 8) + 1),
+        parentTunnel: process.env.SAUCE_TUNNEL || PARENT_TUNNEL,
+        name: process.env.npm_package_name || 'Athenz UI Tests',
+        build: BUILD_NUMBER,
+        screenResolution: functionalConfig.screenResolution
+      }
+    }
+  ];
+}
+let config = {
+    ...localOrRemote.saucelabs,
+    //
+    // ====================
+    // Runner Configuration
+    // ====================
+    // WebdriverIO supports running e2e tests as well as unit and component tests.
+    runner: 'local',
```

Comment:
wouldn't this set runner as local for saucelabs test too?

Reply:
it would be, but this doesn't affect saucelabs functional test execution